xla/service/gpu/gpu_windowed_einsum_handler_test.cc

```diff
@@ -193,5 +197,146 @@ ENTRY main.9_spmd {
       cp1->backend_config<GpuBackendConfig>()->force_earliest_schedule());
 }
 
+TEST_F(GpuWindowedEinsumHanlderTest, AgLoopsMultipleConsumersAreChained) {
```

Comment:
Would it be possible to have a minimal test case?

Reply:
good point, cut down the test case hlo to the minimum.

Reply:
I still have a hard time following this transformation. If it is as simple as depicted in the diagram, can we have an additional test case with just those few operations? I think a hand-crafted HLO test case to document this transformation would really help.

Reply:
So the pass alone is trying to do 2 things:
1. look for an already rewritten windowed einsum loop
2. check if the loop is sharing the same operand with another dot
When these 2 conditions are met, we will perform the transformation in the diagram, so we need to have a rewritten windowed einsum loop in the graph to test this pass. In order to test just the operations in the diagram, we'd need to call the SPMD partitioner in this test. I have written an e2e test for this here `CollectiveOpsTestE2E.WindowedEinsumE2EAllgatherMultiConsumer` under `//xla/tests:collective_ops_test_e2e`